File: 3867-sum-of-gcd-of-formed-pairs/3867-sum-of-gcd-of-formed-pairs.py
class Solution:
    def gcdSum(self, nums: list[int]) -> int:
        res = 0
        n = len(nums)
        mx = 0
        prefixGcd = [gcd(max(mx, num), num) for num in nums if (mx := max(mx, num))]
        prefixGcd.sort()
        for i in range(n // 2):
            res += gcd(prefixGcd[i], prefixGcd[n - i - 1])
        return res

File: 3867-sum-of-gcd-of-formed-pairs/3867-sum-of-gcd-of-formed-pairs.cpp
class Solution {
public:
    long long gcdSum(vector<int>& nums) {
        long long res = 0;
        int n = nums.size();
        int mx = 0;
        vector<int> prefixGcd;
        for(int i = 0; i < n; i++)
        {
            mx = max(mx, nums[i]);
            prefixGcd.push_back(gcd(nums[i], mx));
        }
        sort(prefixGcd.begin(), prefixGcd.end());
        for(int i = 0; i < n / 2; i++)
        {
            res += gcd(prefixGcd[i], prefixGcd[n - i - 1]);
        }
        return res;
    }
};